src/aof.c

```diff
@@ -2296,7 +2310,7 @@ int rewriteAppendOnlyFileBackground(void) {
         redisSetProcTitle("redis-aof-rewrite");
         redisSetCpuAffinity(server.aof_rewrite_cpulist);
         snprintf(tmpfile,256,"temp-rewriteaof-bg-%d.aof", (int) getpid());
-        if (rewriteAppendOnlyFile(tmpfile) == C_OK) {
+        if (rewriteAppendOnlyFile(tmpfile, RDBFLAGS_AOF_PREAMBLE) == C_OK) {
```

Comment:
ohh, i now see this code in `startSaving`.
```c
    else if (getpid()!=server.pid)
        subevent = REDISMODULE_SUBEVENT_PERSISTENCE_RDB_START;
    else
        subevent = REDISMODULE_SUBEVENT_PERSISTENCE_SYNC_RDB_START;
```
so i suppose we can do the same for AOF (in `startLoading`) and there's no need to pass any flags to `rewriteAppendOnlyFile`, and no need to change the input of `startSaving` (it can just use the right module event depending on the pid).
i.e. no need for the change in rdb.h, and no need for the flags argument for rewriteAppendOnlyFile

Reply:
I don't understand what you mean, we are generating a persist event, not a loading event

Reply:
sorry, that was a typo. i meant `startSaving`

Reply:
good suggestion.